### ISSUE TITLE
Extend support for {rms}

### ISSUE THREAD

**Issue by Hjalt** [Rapporter/pander#207]
Would it be possible to include support for the model objects (lrm, orm etc.) in Frank Harrell's rms package? In itself it supports LaTeX output for tables, but I would love to see support for output in markdown.

**Comment by daroczig** [comment]
Sure! Can you please provide a list of the model objects with some examples how to generate those?

**Comment by Hjalt** [comment]
All right, let's see. Here's an example covering a model fit with the `ols` function in the {rms} package. It should be noted that there are other functions, such as `lrm`, `orm`, `rms`, `Glm`, `cph`, and their output varies slightly.
```r
require(rms)

set.seed(123)
n <- 1000
age <- rnorm(n, 50, 10)
cholesterol <- rnorm(n, 200, 25)
sex <- factor(sample(c("female", "male"), n, TRUE))
health <- data.frame(age, cholesterol)

dd <- datadist(age, sex) # Is needed for summary() below
options(datadist = 'dd')    

fit1 <- ols(cholesterol ~ age)
fit2 <- ols(cholesterol ~ age + sex)
```

```r
# lrtest() can be used to print a model comparison based on likelihood ratio 
lrtest(fit1, fit2)

Model 1: cholesterol ~ age
Model 2: cholesterol ~ age + sex

L.R. Chisq       d.f.          P 
0.2224600  1.0000000  0.6371719
```

```r
# Default print method
fit2

Linear Regression Model

ols(formula = cholesterol ~ age + sex)

                 Model Likelihood     Discrimination    
                    Ratio Test           Indexes        
Obs      1000    LR chi2      7.73    R2       0.008    
sigma 25.1697    d.f.            2    R2 adj   0.006    
d.f.      997    Pr(> chi2) 0.0210    g        2.501    

Residuals

      Min        1Q    Median        3Q       Max 
-75.32269 -17.42416  -0.05717  17.86732  82.67025 

          Coef     S.E.   t     Pr(>|t|)
Intercept 190.4515 4.2067 45.27 <0.0001 
age         0.2188 0.0803  2.72 0.0066  
sex=male   -0.7503 1.5932 -0.47 0.6378  
```
```r
# Summary
summary(fit2)

             Effects              Response : cholesterol 

 Factor            Low    High   Diff.  Effect   S.E.   Lower 0.95 Upper 0.95
 age               43.717 56.646 12.929  2.82940 1.0388  0.79086   4.8679    
 sex - male:female  1.000  2.000     NA -0.75034 1.5932 -3.87670   2.3760
```

```r
# Summary can be coerced into a data frame, but that adds an (unwanted) new column, 'type'
data.frame(summary(fit2))

                       Low     High    Diff.     Effect     S.E. Lower.0.95 Upper.0.95 Type
age               43.71676 56.64602 12.92926  2.8293870 1.038820  0.7908622   4.867912    1
sex - male:female  1.00000  2.00000       NA -0.7503439 1.593172 -3.8766988   2.376011    1
```
```r
# Analysis of variance table with F tests for predictors
anova(fit2)

                Analysis of Variance          Response: cholesterol 

 Factor     d.f. Partial SS  MS        F    P     
 age          1    4699.5907 4699.5907 7.42 0.0066
 sex          1     140.5243  140.5243 0.22 0.6378
 REGRESSION   2    4900.8181 2450.4091 3.87 0.0212
 ERROR      997  631613.2488  633.5138
```     

Most of these print functions seem to be passed to `prModFit´ from ´Hmisc´. From the [vignette](https://cran.r-project.org/web/packages/rms/rms.pdf):

> prModFit is the workhorse for the print methods for highest-level rms model fitting functions, handling both regular and LaTeX printing, the latter resulting in LaTeX code written to the terminal,
automatically ready for Sweave. The work of printing summary statistics is done by prStats, which
uses the Hmisc print.char.matrix function to print overall model statistics if latex=FALSE, otherwise
it generates customized LaTeX code. The LaTeX longtable and epic packages must be in effect to use these LaTeX functions.

I hope this is what you wanted. If you want examples for some of the other functions, tell me!

**Comment by RomanTsegelskyi** [comment]
Thanks for an extended example. I will work on it this week. `rms` has an interesting system for printing, where it `print.ols` (and other `print.*` functions), transform to some kind of intermediate representation to be passed to `prModFit´ from ´Hmisc´. I will try to look into how to print that IR using `pander`'s existing rendering facilities.

**Comment by RomanTsegelskyi** [comment]
I implemeted a test version for `ols`. 

```r
> pander(fit1)

-------------------------------------------------------------------------
 &nbsp;   &nbsp;    &nbsp;    Model Likelihood   &nbsp;   Discrimination 
                                 Ratio Test                  Indexes     
-------- -------- ---------- ------------------ -------- ----------------
  Obs    1000.00   LR chi2        7.506799         R2        0.007479    

 sigma    25.16      d.f.         1.000000       R2 adj      0.006484    

  d.f.    998.00  Pr(> chi2)      0.006147         g         2.463815    
-------------------------------------------------------------------------

Table: Fitting linear model: cholesterol ~ age


---------------------------------
 Min    1Q    Median   3Q    Max 
----- ------ -------- ----- -----
-75.7 -17.29  0.1067  17.72 82.28
---------------------------------

Table: Residuals


----------------------------------------------------
    &nbsp;        Coef    S.E.    Wald Z   Pr(>|Z|) 
--------------- -------- ------- -------- ----------
 **Intercept**  190.0202 4.10425  46.298      0     

    **age**      0.2201  0.08027  2.742       0     
----------------------------------------------------

Table: Coeficients
```
It turned out a bit harder than I expected, because or complicated printing facilities in `rms`. `rms` delegates printing to `prModFit`, which in turn has different printing functions for different tables (like `prStats`, etc.). 

Only thing I am not sure about is first table. Markdown doesn't support rowspan, so I am not sure how is best to recreate 

```r
ols(formula = cholesterol ~ age + sex)

                 Model Likelihood     Discrimination    
                    Ratio Test           Indexes        
Obs      1000    LR chi2      7.73    R2       0.008    
sigma 25.1697    d.f.            2    R2 adj   0.006    
d.f.      997    Pr(> chi2) 0.0210    g        2.501  
``` 
in markdown. @Hjalt @daroczig, what do you think?

**Comment by daroczig** [comment]
Thanks a lot, @RomanTsegelskyi!

This is how the table renders in pdf:

![image](https://cloud.githubusercontent.com/assets/495736/8923403/6e81f6da-34a5-11e5-8a43-87ead69d5862.png)

It looks a bit strange, probably removing the space in the `Pr(> chi2)` cell would improve it a bit:

![image](https://cloud.githubusercontent.com/assets/495736/8923454/d96ad700-34a5-11e5-98f4-92eab0d9f0a9.png)

Still not the best :( Not sure why we have that extra padding after the first line. In HTML it looks IMO good:

![image](https://cloud.githubusercontent.com/assets/495736/8923453/d2c7af72-34a5-11e5-81b0-2d012250ce37.png)

Will do the code review in the pull request. Thanks again!

**Comment by Hjalt** [comment]
Very nice work! I actually do not have any issues with whitespace in my pdf output.

![](https://github.com/Hjalt/pander/blob/master/ols-default-test1.png?raw=true)

However, I think that the first table is currently somewhat missleading. It is in fact three different tables, where the first one give the number of observations, sigma, and degrees of freedom, whereas the second (which is now the header for column number 4) is the model likelihood ratio test, which should span the third and fourth c olumns. Finally, the third table (fifth and sixth columns are the discrimination indexes, that is, the R2, R2 adjusted, and g values. 

Do you see what I mean? If I understand correctly, there's no support for multi-column cells at the moment, but perhaps it could be accomplished if the table was transposed and a multi-line solution was adopted? Alternatively, the table could simply be split in three. 

I'm sorry that I can't be of any actual help in accomplishing this!

**Comment by RomanTsegelskyi** [comment]
@Hjalt, I totally agree, and that's why I am not completely satisfied with the first table. The only other idea I had was to break it into 3 different table, but that just blow output too much.

**Comment by daroczig** [comment]
Not sure what's broken with my `pdflatex` then -- but I'm happy to see that the implementation is working fine :)

Anyway, to be constructive: what if we add an extra (~empty) column between the 3 tables to technically keep those in the same table, yet separate those a bit from each other?

Another idea might be to have only 3 columns, and not splitting the keys and values into two columns, but each cell could include the key and value as well, such as `df: 2` -- all left aligned.

And probably the least constructive question: do we really need to present all those values?

**Comment by Hjalt** [comment]
Yeah, I had that in mind too but I agree that its a lot of information. I think there is a good reason to present all the values. Even if I don't expect people to publish a paper with all that, for the most part they're valuable diagnostics of model fit and useful if you're writing up a lab report.

I vote for the second solution that you're proposing, i.e. combine the columns. It seems clear and tidy enough.

**Comment by RomanTsegelskyi** [comment]
Here is implementation of idea with empty column
```r
----------------------------------------------------------------------------------------------
 &nbsp;   &nbsp;   &nbsp;   Model Likelihood   Ratio Test   &nbsp;   Discrimination   Indexes 
-------- -------- -------- ------------------ ------------ -------- ---------------- ---------
  Obs    1000.00                LR chi2         7.506799                   R2        0.007479 

 sigma    25.16                   d.f.          1.000000                 R2 adj      0.006484 

  d.f.    998.00               Pr(> chi2)       0.006147                   g         2.463815 
----------------------------------------------------------------------------------------------

Table: Fitting linear model: cholesterol ~ age
```

And here is the result somewhat similar to second approach that Gergely described (values are below the statistic)
```r
--------------------------------------------------------
     &nbsp;       Model Likelihood     Discrimination   
                     Ratio Test            Indexes      
---------------- ------------------- -------------------
      Obs              LR chi2               R2         

      1000        7.50679944718909   0.00747869380001054

     sigma              d.f.               R2 adj       

25.1598849915534          1          0.00648418347315682

      d.f.           Pr(> chi2)               g         

      998        0.00614664988798697  2.46381529794586  
--------------------------------------------------------
```
In that case, rounding and formatting of numbers needs to be performed before `pandoc.table`, which will "override" `pandoc.table` formatting functionality. I am a bit torn, though second approach seems a bit better. Any other thoughts?

**Comment by daroczig** [comment]
The empty column option looks pretty cool in PDF, but the other seems to be similarly good to me. I leave the decision on you, guys. Thank you @RomanTsegelskyi for the examples!

About the rounding/digits pre-processing: it shouldn't be that hard to apply the `panderOptions` on the numbers -- either by calling `format` in the S3 method, or applying `p` with `wrap=''`.

**Comment by Hjalt** [comment]
They are both pretty neat! I think you should choose whichever options seems easier to implement for you.

**Comment by RomanTsegelskyi** [comment]
In the end I went for second choice. I will try to implement support for other functions soon - #212

**Comment by RomanTsegelskyi** [comment]
Merged #212
